src/haystack/analyzer.clj

```diff
@@ -120,17 +120,26 @@
     (flag-frame frame :repl)
     frame))
 
+(defn- tool? [frame-name last?]
+  (boolean (or (re-find #"^clojure\.lang\.AFn|^clojure\.lang\.RestFn|^clojure\.lang\.RT|clojure\.lang\.Compiler|^nrepl\.|^cider\.|^clojure\.core/eval|^clojure\.core/apply|^clojure\.core/with-bindings|^clojure\.core/binding-conveyor-fn|^clojure\.main/repl"
```

Comment:
I tried this and it will need a slight tweak because real `frame-name`s are munged:

```
["clojure.main$repl$read_eval_print__9234$fn__9235/invoke"
   "clojure.main$repl$read_eval_print__9234/invoke"
   "clojure.main$repl$fn__9243/invoke"
   "clojure.main$repl/invokeStatic"
   "clojure.main$repl/doInvoke"
   "clojure.lang.RestFn/invoke"
   "nrepl.middleware.interruptible_eval$evaluate/invokeStatic"
   "nrepl.middleware.interruptible_eval$evaluate/invoke"
   "nrepl.middleware.interruptible_eval$interruptible_eval$fn__384954$fn__384969/invoke"
   "clojure.lang.AFn/run"
   "nrepl.middleware.session$session_exec$main_loop__385762$fn__385779/invoke"
   "nrepl.middleware.session$session_exec$main_loop__385762/invoke"
   "clojure.lang.AFn/run"
   "java.lang.Thread/run"
   "clojure.lang.LispReader/read"
   "clojure.lang.LispReader/read"
   "clojure.lang.LispReader/read"
   "clojure.core$read/invokeStatic"
   "clojure.core$read/invoke"
   "nrepl.middleware.interruptible_eval$evaluate$fn__384007/invoke"
   "clojure.main$repl$read_eval_print__9234$fn__9235/invoke"
   "clojure.main$repl$read_eval_print__9234/invoke"
   "clojure.main$repl$fn__9243/invoke"
   "clojure.main$repl/invokeStatic"
   "clojure.main$repl/doInvoke"
   "clojure.lang.RestFn/invoke"
   "nrepl.middleware.interruptible_eval$evaluate/invokeStatic"
   "nrepl.middleware.interruptible_eval$evaluate/invoke"
   "nrepl.middleware.interruptible_eval$interruptible_eval$fn__384954$fn__384969/invoke"
   "clojure.lang.AFn/run"
   "nrepl.middleware.session$session_exec$main_loop__385762$fn__385779/invoke"
   "nrepl.middleware.session$session_exec$main_loop__385762/invoke"
   "clojure.lang.AFn/run"
   "java.lang.Thread/run"
   "clojure.lang.Util/runtimeException"
   "clojure.lang.LispReader/interpretToken"
   "clojure.lang.LispReader/read"
   "clojure.lang.LispReader/read"
   "clojure.lang.LispReader/read"
   "clojure.core$read/invokeStatic"
   "clojure.core$read/invoke"
   "nrepl.middleware.interruptible_eval$evaluate$fn__384007/invoke"
   "clojure.main$repl$read_eval_print__9234$fn__9235/invoke"
   "clojure.main$repl$read_eval_print__9234/invoke"
   "clojure.main$repl$fn__9243/invoke"
   "clojure.main$repl/invokeStatic"
   "clojure.main$repl/doInvoke"
   "clojure.lang.RestFn/invoke"
   "nrepl.middleware.interruptible_eval$evaluate/invokeStatic"
   "nrepl.middleware.interruptible_eval$evaluate/invoke"
   "nrepl.middleware.interruptible_eval$interruptible_eval$fn__384954$fn__384969/invoke"
   "clojure.lang.AFn/run"
   "nrepl.middleware.session$session_exec$main_loop__385762$fn__385779/invoke"
   "nrepl.middleware.session$session_exec$main_loop__385762/invoke"
   "clojure.lang.AFn/run"
   "java.lang.Thread/run"
   "clojure.lang.Numbers/divide"
   "clojure.lang.Numbers/divide"
   "cider.nrepl.inlined.deps.haystack.v0v1v0.haystack.analyzer$eval1308394/invokeStatic"
   "cider.nrepl.inlined.deps.haystack.v0v1v0.haystack.analyzer$eval1308394/invoke"
   "clojure.lang.Compiler/eval"
   "clojure.lang.Compiler/eval"
   "clojure.core$eval/invokeStatic"
   "clojure.core$eval/invoke"
   "nrepl.middleware.interruptible_eval$evaluate$fn__383933$fn__383946/invoke"
   "clojure.lang.AFn/applyToHelper"
   "clojure.lang.AFn/applyTo"
   "clojure.core$apply/invokeStatic"
   "clojure.core$with_bindings_STAR_/invokeStatic"
   "clojure.core$with_bindings_STAR_/doInvoke"
   "clojure.lang.RestFn/invoke"
   "nrepl.middleware.interruptible_eval$evaluate$fn__383933/invoke"
   "clojure.main$repl$read_eval_print__9234$fn__9237/invoke"
   "clojure.main$repl$read_eval_print__9234/invoke"
   "clojure.main$repl$fn__9243/invoke"
   "clojure.main$repl/invokeStatic"
   "clojure.main$repl/doInvoke"
   "clojure.lang.RestFn/invoke"
   "nrepl.middleware.interruptible_eval$evaluate/invokeStatic"
   "nrepl.middleware.interruptible_eval$evaluate/invoke"
   "nrepl.middleware.interruptible_eval$interruptible_eval$fn__384954$fn__384969/invoke"
   "clojure.lang.AFn/run"
   "nrepl.middleware.session$session_exec$main_loop__385762$fn__385779/invoke"
   "nrepl.middleware.session$session_exec$main_loop__385762/invoke"
   "clojure.lang.AFn/run"
   "java.lang.Thread/run"]
```